integration-libs/opf/b2b-checkout/components/opf-b2b-checkout-payment-type/opf-b2b-checkout-payment-type.component.spec.ts

```diff
@@ -0,0 +1,175 @@
+  setCartPaymentOption = () => of({} as Cart);
+}
+
+describe('OpfB2bCheckoutPaymentTypeComponent', () => {
+  let component: OpfB2bCheckoutPaymentTypeComponent;
+  let fixture: ComponentFixture<OpfB2bCheckoutPaymentTypeComponent>;
+  let checkoutStepService: CheckoutStepService;
+  let checkoutPaymentTypeFacade: CheckoutPaymentTypeFacade;
+  let activeCartFacade: ActiveCartFacade;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [ReactiveFormsModule, I18nTestingModule],
+      declarations: [OpfB2bCheckoutPaymentTypeComponent, MockTranslatePipe],
+      providers: [
+        { provide: ActiveCartFacade, useClass: MockActiveCartFacade },
+        {
+          provide: CheckoutPaymentTypeFacade,
+          useClass: MockCheckoutPaymentTypeFacade,
+        },
+        { provide: CheckoutStepService, useClass: MockCheckoutStepService },
+        {
+          provide: OpfMetadataStoreService,
+          useClass: MockOpfMetadataStoreService,
+        },
+        { provide: OpfPaymentFacade, useClass: MockOpfPaymentFacade },
+        { provide: UserIdService, useValue: {} },
+        { provide: GlobalMessageService, useValue: {} },
+        { provide: ActivatedRoute, useValue: {} },
+      ],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(OpfB2bCheckoutPaymentTypeComponent);
+    component = fixture.componentInstance;
+    checkoutStepService = TestBed.inject(CheckoutStepService);
+    checkoutPaymentTypeFacade = TestBed.inject(CheckoutPaymentTypeFacade);
+    activeCartFacade = TestBed.inject(ActiveCartFacade);
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should handle payment change', () => {
+    const payment: OpfActiveConfiguration = {
+      id: 1,
+      paymentType: B2BPaymentTypeEnum.ACCOUNT_PAYMENT,
+      providerType: OpfPaymentProviderType.PAYMENT_GATEWAY,
+      displayName: 'Test Payment',
+      description: 'Test Description',
+    };
+    spyOn(checkoutStepService, 'disableEnableStep');
+    spyOn(checkoutPaymentTypeFacade, 'setPaymentType').and.returnValue(of({}));
+    spyOn(activeCartFacade, 'reloadActiveCart');
+
+    component.handlePaymentChange(payment);
+
+    expect(checkoutStepService.disableEnableStep).toHaveBeenCalledWith(
+      CheckoutStepType.PAYMENT_DETAILS,
+      true
+    );
+    expect(checkoutStepService.disableEnableStep).toHaveBeenCalledWith(
+      CheckoutStepType.REVIEW_ORDER,
+      false
+    );
+    expect(checkoutPaymentTypeFacade.setPaymentType).toHaveBeenCalled();
+    expect(activeCartFacade.reloadActiveCart).toHaveBeenCalled();
+  });
+
+  it('should navigate to next step', () => {
+    spyOn(checkoutStepService, 'next');
+    spyOn(checkoutPaymentTypeFacade, 'setPaymentType').and.returnValue(of({}));
+    spyOn(activeCartFacade, 'reloadActiveCart');
+
+    // Set required state
+    component['selectedPaymentOption'] = '1';
+    component['poNumberInputElement'] = {
```

Comment:
It's not great to work with private properties in tests.

Reply:
Right, I changed a test and component logic a bit and now we are using public method to update component state in test.